3D/senior-csharp-test/Assets/Cache/FileCacheStorage.cs

```diff
@@ -52,218 +28,121 @@ private void InitCache()
 			return;
 		}
 
-		if( TryLoadCacheEntries() )
-		{
-			CheckCacheVersion();
-			ClearExpiredEntries();
-		}
-		else
-			DeleteAll();
+		ClearExpiredCacheFiles();
 	}
 
-	private bool TryLoadCacheEntries()
+	private void ClearExpiredCacheFiles()
 	{
-		string filePath = _cachePath + "/" + _fileName;
+		List<string> expiredfilePath = GetExpiredFilePaths();
 
-		if( !_fileSystem.FileExists( filePath ) )
-			return false;
-
-		try
+		foreach( string filePath in expiredfilePath )
 		{
-			LoadCacheEntriesFromFile( filePath );
-			return true;
-		}
-		catch( Exception e )
-		{
-			Log.LogError( "Could not load cache entries from " + filePath + "!\n" + e.Message + "\n" + e.StackTrace );
+			try
+			{
+				_fileSystem.DeleteFile( filePath );
+			}
+			catch( Exception e )
+			{
+				Log.LogError( "Fail to delete expired cache file at path : " + filePath + ".\n" + e );
+			}
 		}
-
-		return false;
 	}
 
-	private void LoadCacheEntriesFromFile( string path )
+	private List<string> GetExpiredFilePaths()
 	{
-		string txt = _fileSystem.ReadText( path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite );
-		_serializer.Deserialize( txt, this );
-		CopyDeserializedEntriesToDic();
-	}
+		DateTime dateNow = DateTime.UtcNow;
 
-	private void CopyDeserializedEntriesToDic()
-	{
-		_cacheEntries.Clear();
-		foreach( FileCacheEntry cacheEntry in CacheEntries )
-			_cacheEntries.Add( cacheEntry.Id, cacheEntry );
-	}
+		List<string> expiredFilePaths = new List<string>();
 
-	private void CheckCacheVersion()
-	{
-		if( Version != ACTUAL_VERSION )
-			DeleteAll();
-	}
+		string[] filePaths = _fileSystem.GetFiles( _cachePath );
 
-	private void ClearExpiredEntries()
-	{
-		List<FileCacheEntry> expiredEntries = GetExpiredEntries();
-
-		foreach( FileCacheEntry t in expiredEntries )
-			Delete( t.Id );
-	}
-
-	private List<FileCacheEntry> GetExpiredEntries()
-	{
-		DateTime dateNow = DateTime.Now;
-		List<FileCacheEntry> expiredEntries = new List<FileCacheEntry>();
-
-		foreach( KeyValuePair<string, FileCacheEntry> ce in _cacheEntries )
+		foreach( string filePath in filePaths )
 		{
-			DateTime datetime = GetEntryLastUseDate( ce.Value );
-			TimeSpan span = dateNow - datetime;
+			DateTime lastAccessDate;
+			try
+			{
+				lastAccessDate = _fileSystem.GetFileLastAccessTime( filePath );
+			}
+			catch( Exception e )
+			{
+				Log.LogError( "Couldn't access file last access time at path = " + filePath + ".\n" + e );
+				continue;
+			}
+
+			TimeSpan span = dateNow - lastAccessDate;
 
 			if( span.TotalDays > EXPIRING_TIME_DAYS )
-				expiredEntries.Add( ce.Value );
+				expiredFilePaths.Add( filePath );
```

Comment:
we could delete the file directly instead of filling in an intermediate list. Just a detail I guess

Reply:
Indeed, I removed GetExpiredFilePaths and put his code directly inside ClearExpiredCacheFiles